### PR TITLE
Update targeting client and model to include optional contacts field on callouts

### PR DESCRIPTION
**Pull request by abeddow91**
## What does this change?
This PR bumps the targeting client to 1.1.4 which adds an optional contacts field on the callout model. This PR also updates the callout extraction method to extract this field and adds it to the model for DCR. 

There will be a secondary PR required on the DCR side to expect this new field. 

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Review by sophie-macmillan — APPROVED** [review]
Looks good - one non blocking question

File: project/Dependencies.scala [hunk 1]
```diff
@@ -70,7 +70,7 @@ object Dependencies {
   // logback2  to prevent "error: reference to logback is ambiguous;"
 
   val kinesisLogbackAppender = "com.gu" % "kinesis-logback-appender" % "1.4.0"
-  val targetingClient = "com.gu" %% "targeting-client" % "1.1.2"
+  val targetingClient = "com.gu" %% "targeting-client" % "1.1.4"
```
[the remark this answers is not in the document]
Comment (abeddow91) [reply project/Dependencies.scala:73]: Ah, I thought 1.1.4 was the most recent version here? but sounds like I could be wrong?

File: common/app/model/dotcomrendering/pageElements/CalloutExtraction.scala [hunk 2]
```diff
@@ -188,6 +193,8 @@ object CalloutExtraction {
       tagName <- (campaign \ "fields" \ "tagName").asOpt[String]
       formFields1 <- (campaign \ "fields" \ "formFields").asOpt[JsArray]
     } yield {
+      val contacts = (campaign \ "fields" \ "contacts").asOpt[Seq[Contact]]
```
Comment (marjisound) [inline common/app/model/dotcomrendering/pageElements/CalloutExtraction.scala:196]: Very interesting that we don't need to do the parsing with the primitive values and `Contact` can be used directly. Whatever example I find online with `asOpt` and also within frontend, all are using it with the innermost primitive fields such as String, Long, etc. But what you did seems to work perfectly 👍

**Review by marjisound — APPROVED** [review]
Looks good, great work 👏

**Pull request merged**